extensions/target-specific/chromium/blink/wtf-helpers/wtf-helpers.js

```diff
@@ -21,7 +21,7 @@ Loader.OnLoad(function() {
     }));
 
     DbgObject.AddTypeDescription(Chromium.RendererProcessType("WTF::StringImpl"), "Text", true, UserEditableFunctions.Create((wtfStringImpl) => {
-        return !wtfStringImpl.isNull() ? wtfStringImpl.idx(1).as("char", /*disregardSize*/true).string(wtfStringImpl.f("length_")) : "";
+        return !wtfStringImpl.isNull() ? wtfStringImpl.idx(1).as("unsigned char", /*disregardSize*/true).string(wtfStringImpl.f("length_")) : "";

```

Comment:
I guess another option would be to change `string` to do use `uvals` unconditionally to make it more forgiving? Especially if `String.fromCharCode` assumes a certain size when given negative numbers.

Reply:
Sorry for all the questions — does `WTF::StringImpl` use UTF-8? Won’t this still be broken?

Reply:
For the first question...open it doing to unconditionally, but just fixing isUnsigned solves the problem here.

For the second question...no, WTF::StringImpl uses Latin-1 or UTF-16 (https://chromium.googlesource.com/chromium/src/+/HEAD/third_party/blink/renderer/platform/wtf/text/README.md). The current implementation of this helper doesn't differentiate between the two, which might be a bug, but I haven't yet found a case with LCHARs or UCHARs where this doesn't work.

Reply:
Should be fixed now.

Reply:
If UChar is 16-bits why is this code doing `.as("char")` at all?  Maybe it has the appearance of working because for ASCII characters it just ends up interpolating a bunch of null characters which don't get rendered?

I'd think that for either case if the `DbgObject` is correctly typed `.string(correctLengthInCountNotBytes)` would just work...

Reply:
It needs to do .as() because the string is stored "inline" in the memory right after the StringImpl. That memory needs to be interpreted as either 8-bit or 16-bit depending on whether StringImpl::is_8bit_ is true.

Reply:
Right, LCHARs are 8-bit unsigned chars, whereas UCHARs are 16-bit.